src/plugins/cosmos/CosmosAsset.tsx

```diff
@@ -0,0 +1,40 @@
+import { Flex } from '@chakra-ui/react'
+import { useParams } from 'react-router-dom'
+import { Page } from 'components/Layout/Page'
+import { LoadingAsset } from 'pages/Assets/LoadingAsset'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import {
+  selectAssetByCAIP19,
+  selectMarketDataById,
+  selectMarketDataLoadingById
+} from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { CosmosAssetAccountDetails } from './CosmosAssetAccountDetails'
+
+export const CosmosAsset = (props: { chainId: string }) => {
+  const dispatch = useAppDispatch()
+
+  const { assetSubId } = useParams<{ assetSubId: string }>()
+  const assetId = `${props.chainId}/${assetSubId}`
+  const asset = useAppSelector(state => selectAssetByCAIP19(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+
+  if (!marketData) dispatch(marketApi.endpoints.findByCaip19.initiate(assetId))
+
+  const loading = useAppSelector(state => selectMarketDataLoadingById(state, assetId))
+
+  return !asset || loading ? (
+    <Page style={{ flex: 1 }} key={asset?.tokenId}>
```

Comment:
Note: If `asset` is undefined, the `key` will be undefined and react will be very sad and throwing errors and warnings. I'd use the constructed `assetId` to avoid that, which is guaranteed to exist and be unique.

Reply:
this isn't a final version of this file. I just put in enough to test the routing working. it doesn't matter that this is broken because prod won't have cosmos assets and the plugin will be disabled.